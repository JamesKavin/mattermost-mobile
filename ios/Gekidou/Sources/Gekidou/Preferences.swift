//
//  Preferences.swift
//  Gekidou
//
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
//
import Foundation

let appGroupId = Bundle.main.infoDictionary!["AppGroupIdentifier"] as? String

public class Preferences: NSObject {
    private let userDefauls = UserDefaults(suiteName: appGroupId)
    
    @objc public static let `default` = Preferences()
    
    override private init() {}
    
    public func object(forKey name: String) -> Any? {
        return userDefauls?.object(forKey: name)
    }
    
    public func removeObject(forKey name: String) {
        userDefauls?.removeObject(forKey: name)
    }
    
    public func set(_ value: Any?, forKey key: String) {
        userDefauls?.set(value, forKey: key)
    }
}
