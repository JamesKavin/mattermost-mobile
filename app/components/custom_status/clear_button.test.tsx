// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import ClearButton from '@components/custom_status/clear_button';
import Preferences from '@constants/preferences';
import {fireEvent, render} from '@test/intl-test-helper';

describe('components/custom_status/clear_button', () => {
    const baseProps = {
        handlePress: jest.fn(),
        testID: 'clear_custom_status.button',
        theme: Preferences.THEMES.denim,
    };

    it('should match snapshot', () => {
        const wrapper = render(
            <ClearButton
                {...baseProps}
            />,
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should call handlePress when press event is fired', () => {
        const {getByTestId} = render(
            <ClearButton
                {...baseProps}
            />,
        );

        fireEvent.press(getByTestId(baseProps.testID));
        expect(baseProps.handlePress).toBeCalled();
    });
});
