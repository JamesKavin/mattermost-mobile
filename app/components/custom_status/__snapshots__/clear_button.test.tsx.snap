// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/custom_status/clear_button should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 40,
    }
  }
  testID="clear_custom_status.button"
>
  <Icon
    name="close-circle"
    size={20}
    style={
      {
        "borderRadius": 1000,
        "color": "rgba(63,67,80,0.52)",
      }
    }
  />
</View>
`;
